src/darker/tests/test_multiline_strings.py

```diff
@@ -0,0 +1,124 @@
+            '"""triple-double-quoted',  # line 37
+            '   two-line string"""',
+            'f"""triple-double-quoted',  # line 39
+            '    two-line f-string"""',
+            'r"""triple-double-quoted',  # line 41
+            '    two-line raw string"""',
+            'rf"""triple-double-quoted',  # line 43
+            '     two-line raw f-string"""',
+            'b"""triple-double-quoted',  # line 45
+            '    two-line bytestring"""',
+            'rb"""triple-double-quoted',  # line 47
+            '     two-line raw bytestring"""',
+            '"""triple-',  # line 49
+            "   double-",
+            "   quoted",
+            "   six-",
+            "   line",
+            '   string"""',  # line 54
+        ]
+    )
+
+    result = multiline_strings.get_multiline_string_ranges(content)
+
+    assert result == [
+        # 1-based, end-exclusive
+        (13, 15),
+        (15, 17),
+        (17, 19),
+        (19, 21),
+        (21, 23),
+        (23, 25),
+        (37, 39),
+        (39, 41),
+        (41, 43),
+        (43, 45),
+        (45, 47),
+        (47, 49),
+        (49, 55),
+    ]
+
+
+# End-exclusive
+TEST_RANGES = [(2, 2), (5, 6), (9, 11), (14, 17)]
+
+
+@pytest.mark.kwparametrize(
+    # `(start, end)` and `ranges` are end-exclusive
+    dict(start=0, end=0, ranges=[], expect=None),
+    dict(start=0, end=42, ranges=[], expect=None),
+    dict(start=0, end=0, ranges=TEST_RANGES, expect=None),
+    dict(start=1, end=2, ranges=TEST_RANGES, expect=None),
+    dict(start=2, end=2, ranges=TEST_RANGES, expect=None),
+    dict(start=1, end=3, ranges=TEST_RANGES, expect=(2, 2)),
+    dict(start=2, end=3, ranges=TEST_RANGES, expect=None),
+    dict(start=3, end=3, ranges=TEST_RANGES, expect=None),
+    dict(start=4, end=5, ranges=TEST_RANGES, expect=None),
+    dict(start=5, end=5, ranges=TEST_RANGES, expect=None),
+    dict(start=4, end=6, ranges=TEST_RANGES, expect=(5, 6)),
+    dict(start=5, end=6, ranges=TEST_RANGES, expect=(5, 6)),
+    dict(start=6, end=6, ranges=TEST_RANGES, expect=None),
+    dict(start=4, end=7, ranges=TEST_RANGES, expect=(5, 6)),
+    dict(start=5, end=7, ranges=TEST_RANGES, expect=(5, 6)),
+    dict(start=6, end=7, ranges=TEST_RANGES, expect=None),
+    dict(start=10, end=10, ranges=TEST_RANGES, expect=(9, 11)),
+    dict(start=10, end=11, ranges=TEST_RANGES, expect=(9, 11)),
+    dict(start=10, end=12, ranges=TEST_RANGES, expect=(9, 11)),
+    dict(start=11, end=11, ranges=TEST_RANGES, expect=None),
+    dict(start=11, end=12, ranges=TEST_RANGES, expect=None),
+    dict(start=12, end=12, ranges=TEST_RANGES, expect=None),
+    dict(start=10, end=19, ranges=TEST_RANGES, expect=(9, 17)),
+)
+def test_find_overlap(start, end, ranges, expect):
+    """`find_overlap()` finds the enclosing range for overlapping ranges"""
+    result = multiline_strings.find_overlap(start, end, ranges)
+
+    assert result == expect
```

Comment:
Maybe, provide a message, to show the values/diff. Easier debugging.

Reply:
But doesn't Pytest already do this for us? If I e.g. change the expected value in the sixth case from `(2, 2)` to `(2, 3)`, Pytest gives me:
```python
$ pytest -k test_find_overlap
================ test session starts =================
platform linux -- Python 3.10.0, pytest-7.0.0, pluggy-1.0.0
rootdir: /home/kaiant/prg/darker, configfile: pytest.ini
plugins: mypy-0.9.0, isort-2.0.0, darker-0.1.2, kwparametrize-0.0.3, anyio-3.5.0, flake8-1.0.7
collected 713 items / 690 deselected / 23 selected                                                                                                                                                                                           

src/darker/tests/test_multiline_strings.py .....F.................                                                                                                                                                                     [100%]

====================== FAILURES ======================
__ test_find_overlap[1-3-ranges5-expect5] __

start = 1, end = 3, ranges = [(2, 2), (5, 6), (9, 11), (14, 17)], expect = (2, 3)

    @pytest.mark.kwparametrize(
        # `(start, end)` and `ranges` are end-exclusive
        # [...]
        dict(start=1, end=3, ranges=TEST_RANGES, expect=(2, 3)),
        # [...]
    )
    def test_find_overlap(start, end, ranges, expect):
        """`find_overlap()` finds the enclosing range for overlapping ranges"""
        result = multiline_strings.find_overlap(start, end, ranges)
    
>       assert result == expect
E       assert (2, 2) == (2, 3)
E         At index 1 diff: 2 != 3
E         Use -v to get the full diff

src/darker/tests/test_multiline_strings.py:124: AssertionError
============== short test summary info ===============
FAILED src/darker/tests/test_multiline_strings.py::test_find_overlap[1-3-ranges5-expect5] - assert (2, 2) == (2, 3)
=======================================
```